droughtpipeline/load.py

```diff
@@ -0,0 +1,872 @@
+from __future__ import annotations
+
+import os.path
+import copy
+import time
+import os
+import json
+import cdsapi  
+
+from droughtpipeline.secrets import Secrets
+from droughtpipeline.settings import Settings
+from droughtpipeline.data import (
+    AdminDataSet,
+    AdminDataUnit,
+    ForecastDataUnit,
+    ThresholdDataUnit,
+    ClimateRegionDataSet,
+    RainfallClimateRegionDataUnit,
+    ClimateRegionDataUnit,
+    RainfallDataUnit,
+    PipelineDataSets,
+)
+from urllib.error import HTTPError
+import urllib.request, json
+from datetime import datetime, timedelta, date
+import azure.cosmos.cosmos_client as cosmos_client
+import logging
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+import requests
+import geopandas as gpd
+from typing import List
+import shutil
+from azure.storage.blob import BlobServiceClient
+from azure.core.exceptions import ResourceNotFoundError
+
+COSMOS_DATA_TYPES = [
+    "climate-region",
+    "seasonal-rainfall-forecast",
+]
+
+
+def get_cosmos_query(
+    start_date=None,
+    end_date=None,
+    country=None,
+    adm_level=None,
+    climate_region_code=None,
+    pcode=None,
+    lead_time=None,
+):
+    query = "SELECT * FROM c WHERE "
+    if start_date is not None:
+        query += f'c.timestamp >= "{start_date.strftime("%Y-%m-%dT%H:%M:%S")}" '
+    if end_date is not None:
+        query += f'AND c.timestamp <= "{end_date.strftime("%Y-%m-%dT%H:%M:%S")}" '
+    if country is not None:
+        query += f'AND c.country = "{country}" '
+    if adm_level is not None:    
+        query += f'AND c.adm_level = "{adm_level}" '
+    if climate_region_code is not None:    
+        query += f'AND c.climate_region_code = "{climate_region_code}" '
+    if pcode is not None:
+        query += f'AND c.adm_level = "{pcode}" '
+    if lead_time is not None:
+        query += f'AND c.adm_level = "{lead_time}" '
+    if query.endswith("WHERE "):
+        query = query.replace("WHERE ", "")
+    query = query.replace("WHERE AND", "WHERE")
+    return query
+
+
+def get_data_unit_id(data_unit: AdminDataUnit, dataset: AdminDataSet):
+    """Get data unit ID"""
+    if hasattr(data_unit, "pcode"):
+        if hasattr(data_unit, "lead_time"):
+            id_ = f"{data_unit.pcode}_{dataset.timestamp.strftime('%Y-%m-%dT%H:%M:%S')}_{data_unit.lead_time}"
+        else:
+            id_ = f"{data_unit.pcode}_{dataset.timestamp.strftime('%Y-%m-%dT%H:%M:%S')}"
+    elif hasattr(data_unit, "climate_region_code"):
+        if hasattr(data_unit, "lead_time"):
+            id_ = f"{data_unit.climate_region_code}_{dataset.timestamp.strftime('%Y-%m-%dT%H:%M:%S')}_{data_unit.lead_time}"
+        else:
+            id_ = f"{data_unit.climate_region_code}_{dataset.timestamp.strftime('%Y-%m-%dT%H:%M:%S')}"
+    else:
+        id_ = f"{dataset.timestamp.strftime('%Y-%m-%dT%H:%M:%S')}"
+    return id_
+
+
+def alert_class_to_threshold(alert_class: str, triggered: bool) -> float:
```

Comment:
It seems you're not actually using this method. (Correctly so, because the logic here should not apply in the same way to drought.) Can you remove it? And check for more unused code to remove?

Reply:
will remove this